syft/frameworks/torch/tensors/interpreters/replicated_shared.py

```diff
@@ -1,39 +1,54 @@
-import random
-from operator import add, sub
+from operator import add, sub, mul
 import torch
 import syft
 from syft.generic.abstract.tensor import AbstractTensor
+from syft.frameworks.torch.mpc.przs import PRZS, gen_alpha_3of3
 
 
 class ReplicatedSharingTensor(AbstractTensor):
     def __init__(
-        self, shares_map=None, owner=None, id=None, tags=None, description=None,
+        self, plain_text=None, players=None, owner=None, id=None, tags=None, description=None,
     ):
         super().__init__(id=id, owner=owner, tags=tags, description=description)
+        self.ring_size = 2 ** 32
+        shares_map = self.__generate_shares_map(plain_text, players)
         self.child = shares_map
-        self.ring_size = 2 ** 5
 
-    def share_secret(self, secret, workers):
+    def __generate_shares_map(self, plain_text, players):
+        if plain_text is None or players is None:
```

Comment:
Q: Shouldn't we better raise an exception here? (maybe something like ```BadValue```)

Reply:
sometimes you need to make an empty tensor like an empty torch tensor, it should be acceptable I think